proc/variables.go

```diff
@@ -80,6 +80,48 @@ type EvalScope struct {
 	CFA    int64
 }
 
+// Represents the address and type information for a variable, the thread field is used to read the variable's value
+type VarAddr struct {
```

Comment:
I had some thoughts on these new types. How about putting this information in `Variable` (as you seemed to hint at in your proposal here (https://groups.google.com/forum/#!topic/delve-dev/Z0lDomkS9LI). I think it is completely reasonable for a variable to know its address, dwarfType, etc.. Also, it seems reasonable for a variable to know what thread it's on. I don't mind paying the cost of evaluating the variable either even if all we want to do is set it (as your other patch would allow), but if it becomes an issue or too expensive then we can investigate passing around flags to prevent actual evaluation, but still return a struct with all other fields filled in, however that seems dangerous (you would assume the `Value` field would be filled in).

That way, in your patch to allow setting variables, you could implement the methods on `Variable` instead of these new types. It also seems very reasonable for a variable to know how to change its value.

Thoughts?